File: Chapter 10/Station.java
package chapter10;

public class Station
{
    private String name;
    
    public Station(String name) {
        this.name = name;
    }
    
    public String getName() {
        return name;
    }
    
    public boolean equals(Object obj) {
        if (obj instanceof Station) {
            Station otherStation = (Station) obj;
            if (otherStation.getName().equalsIgnoreCase(name))
            {
                return true;
            }
        }
        return false;
    }
    
    public int hashCode() {
        return name.toLowerCase().hashCode();
    }
}

File: Ch 10 - The OOA&D Lifecycle/Iteration 2 - Objectville Subway/SubwaySystem/Station.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Objectville_Subway.SubwaySystem
{
    class Station
    {
        public string Name { get; }

        public Station(string name)
        {
            Name = name;
        }

        public int hashCode()
        {
            return Name.ToLower().GetHashCode();
        }

        public override bool Equals(object obj)
        {
            if (obj is Station)
            {
                Station otherStation = (Station)obj;
                if (string.Compare(otherStation.Name, Name,ignoreCase:true) == 0)
                {
                    return true;
                }
            }

            return false;
        }

        public override int GetHashCode()
        {
            return Name.ToLower().GetHashCode();
        }
    }
}
